src/features/applications/components/ApplicationsToApprove.tsx

```diff
@@ -146,18 +150,32 @@ type TApplicationsToApprove = z.infer<typeof ApplicationsToApproveSchema>;
 export const ApplicationsToApprove = (): JSX.Element => {
   const applications = useApplications();
   const approved = useApprovedApplications();
-  const approve = useApproveApplication({});
+  const approve = useApproveApplication();
+  const [refetchedData, setRefetchedData] = useState<Attestation[]>();
 
   const approvedById = useMemo(
     () =>
-      approved.data?.reduce((map, x) => {
+      [...(approved.data ?? []), ...(refetchedData ?? [])].reduce((map, x) => {
         map.set(x.refUID, true);
         return map;
       }, new Map<string, boolean>()),
-    [approved.data],
+    [approved.data, refetchedData],
   );
 
-  const applicationsToApprove = applications.data?.filter((application) => !approvedById?.get(application.id));
+  const applicationsToApprove = applications.data?.filter((application) => !approvedById.get(application.id));
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const ret = await fetchApprovedApplications();
+      setRefetchedData(ret);
+    };
+
+    /// delay refetch data for 5 seconds
+    const timeout = setTimeout(() => {
+      fetchData();
+      clearTimeout(timeout);
+    }, 5000);
```

Comment:
```suggestion
    const timeout = setTimeout(() => {
      fetchData();
    }, 5000);
    
    return () => clearTimeout(timeout);
```

Reply:
I also wanna do this, but the linter kept on complaining 🤔